addons/website/models/website.py

```diff
@@ -1767,12 +1771,27 @@ def _trigram_enumerate_words(self, search_details, search, limit):
             ]
             indirect_similarities = []
             for field_info in indirect_fields.values():
+                direct = field_info['direct']
+                direct_field = model._fields[direct]
                 comodel = field_info['comodel']
-                coalias = query.make_alias(model._table, field_info['direct'])
-                query.add_join('LEFT JOIN', coalias, comodel._table, SQL("%s = %s",
-                    SQL.identifier(model._table, 'id'),
-                    SQL.identifier(coalias, field_info['cofield']),
-                ))
+                coalias = query.make_alias(model._table, direct)
+                cofield = field_info['cofield']
+                if cofield:
+                    query.add_join('LEFT JOIN', coalias, comodel._table, SQL("%s = %s",
+                        SQL.identifier(model._table, 'id'),
+                        SQL.identifier(coalias, cofield),
+                    ))
+                elif 'relation' in dir(direct_field):
```

Comment:
It is okay with me :slightly_smiling_face: I tested and it works well :+1: 

Maybe just add a comment here explaining that it is for Many2Many fields 
It would be cool to explain here or in the commit message what exactly is being done because it is complicated to understand 
Here I understand because you explained it to me IRL but it is not crystal clear (it is hard to visualize, but maybe it is because I am bad with all the python/db side...) but since it works, I guess it is alright